src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,23 +38,34 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String latString = ""; 
+    String lngString = "";
+    String product = "";
+    
     try {
-    String latString = request.getParameter("lat");
-    String lngString = request.getParameter("lng");
-    double lat = convertToDouble(latString);
-    double lng = convertToDouble(lngString);
-    MapLocation userLocation = new MapLocation(lat, lng);
-    smallCityService.createUserServiceWithGeolocation(userLocation);
-    } catch(NullPointerException e) {
-      String zip = request.getParameter("zipCode");
-      smallCityService.createUserServiceWithZip(zip);
+      latString = request.getParameter("lat");
+      lngString = request.getParameter("lng");
+      product = request.getParameter("product");
+    } catch (NullPointerException e) {
       LOGGER.warning(e.getMessage() 
            + "Unable to geolocate user, zipCode entered instead.");
+    } finally {
+      SearchObject searchObject = new SearchObject(product);
+      if (latString != null && lngString != null) {
```

Comment:
This should probably check for null and isEmpty. How does MapLocation work with an empty lat or lng value?

Reply:
I added the extra checks - MapLocation does not do much in itself but I would imagine that the Places API calls would fail with empty coordinates.